### PR TITLE
chore: Replace Twitter & Diaspora links and text

### LINKED ISSUE #48409: Sharing settings: Replace Diaspora and Twitter with Mastodon and X
Twitter -> X
Diaspora -> Mastodon

You can have a look at the firstrunwizard for the correct icons + URLs to use.

image

### PR DESCRIPTION
**Pull request by nfebe**
- Use Twitters new name "X"
- Remove Diaspora social and include Mastodon instead

Resolves : https://github.com/nextcloud/server/issues/48409

## Screenshots

| Before |
| ---------- |
|Screenshot from 2024-10-02 11-09-44  |

| After |
| ------- |
|Screenshot from 2024-10-02 11-48-12|

### CODE REVIEW

File: apps/federatedfilesharing/src/components/PersonalSettings.vue [hunk 1]
```diff
@@ -29,19 +29,19 @@
 					<Facebook :size="20" />
 				</template>
 			</NcButton>
-			<NcButton @click="goTo(shareTwitterUrl)">
-				{{ t('federatedfilesharing', 'Twitter') }}
+			<NcButton @click="goTo(shareXUrl)">
+				{{ t('federatedfilesharing', 'Formerly Twitter') }}
```
Comment (provokateurin) [inline apps/federatedfilesharing/src/components/PersonalSettings.vue:33]:
```suggestion
				{{ t('federatedfilesharing', 'X (formerly Twitter)') }}
```
I think it is preferred to have it this way (or just remove the entire Twitter part).

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: apps/federatedfilesharing/src/components/PersonalSettings.vue [hunk 2]
```diff
@@ -29,19 +29,19 @@
...
 				<template #icon>
-					<Twitter :size="20" />
+					<svg width="20"
```
Comment (artonge) [inline apps/federatedfilesharing/src/components/PersonalSettings.vue:35]: No other way to have the X icon?

File: apps/federatedfilesharing/src/components/PersonalSettings.vue (hunk 1 above)
Comment (szaimen) [reply apps/federatedfilesharing/src/components/PersonalSettings.vue:33]: I think keeping the twitter part is better. Since X is not really a word imo

Comment (provokateurin) [reply apps/federatedfilesharing/src/components/PersonalSettings.vue:33]: Agreed, just shows this rename was stupid

File: apps/federatedfilesharing/src/components/PersonalSettings.vue (hunk 2 above)
Comment (Fenn-CS) [reply apps/federatedfilesharing/src/components/PersonalSettings.vue:35]: I tried a couple they all look really, so I just used a vector of the real x logo

Comment (provokateurin) [reply apps/federatedfilesharing/src/components/PersonalSettings.vue:35]: What about the license of the icon then?

Comment (Fenn-CS) [reply apps/federatedfilesharing/src/components/PersonalSettings.vue:35]: I don't know any to be honest, it's just a copy and paste of the centralized version you made.

File: apps/federatedfilesharing/src/components/PersonalSettings.vue [hunk 3]
```diff
@@ -29,19 +29,19 @@
...
+						height="20"
+						viewBox="0 0 15 15"
+						xmlns="http://www.w3.org/2000/svg"><path fill="black" d="m 3.384891,2.6 c -0.3882,0 -0.61495,0.4362184 -0.39375,0.7558594 L 6.5841098,8.4900156 2.9770785,12.707422 C 2.7436785,12.979821 2.9370285,13.4 3.2958285,13.4 H 3.694266 c 0.176,0 0.3430313,-0.07714 0.4570313,-0.210938 L 7.294266,9.5065156 9.6602817,12.887891 C 9.8762817,13.208984 10.25229,13.4 10.743485,13.4 h 1.900391 c 0.3882,0 0.61575,-0.436688 0.39375,-0.754688 L 9.2466097,7.2195156 12.682547,3.1941408 C 12.881744,2.9601408 12.715528,2.6 12.407473,2.6 h -0.506566 c -0.175,0 -0.34186,0.076453 -0.45586,0.2197656 L 8.5405785,6.2058438 6.3790317,3.1132812 C 6.1568442,2.7913687 5.6965004,2.6 5.3958285,2.6 Z" /></svg>
 				</template>
 			</NcButton>
-			<NcButton @click="goTo(shareDiasporaUrl)">
-				{{ t('federatedfilesharing', 'Diaspora') }}
+			<NcButton @click="goTo(shareMastodonUrl)">
+				{{ t('federatedfilesharing', 'Mastodon') }}
```
Comment (susnux) [inline apps/federatedfilesharing/src/components/PersonalSettings.vue:42]: This is a brand name so we do not need to translate it

```suggestion
				Mastodon
```

Comment (susnux) [reply apps/federatedfilesharing/src/components/PersonalSettings.vue:40]: but not related

**Review by susnux — APPROVED** [review]
Just replaced the icons with the proper ones from `core/img` for legal reasons.

Screenshot 2024-10-07 at 18-14-26 Sharing - Personal settings - Nextcloud
Screenshot 2024-10-07 at 18-14-18 Sharing - Personal settings - Nextcloud

**Review by provokateurin — APPROVED** [review]

**Pull request merged**